core/stdlib/std.ncl

```diff
@@ -172,6 +172,25 @@
         "%
       = fun n l => %elem_at% l n,
 
+    at_or
+      : forall a. Number -> a -> Array a -> a
```

Comment:
The additional contract ensures that only positive natural numbers are provided. Otherwise the error is going to be less nice (a dynamic error from `%elem_at%`). We could also return the default values if the index is negative or isn't an integer, but that honestly feels wrong. So I propose we made it a clear contract error when the number isn't a proper index, and return the default value only when the index is greater than the length of the array.

```suggestion
      : forall a. Number -> a -> Array a -> a
      | std.number.PosNat -> Dyn
```